Tools/clinic/clinic.py

```diff
@@ -5573,12 +5564,10 @@ def add_parameter(text: str) -> None:
 
         return docstring
 
-    def state_terminal(self, line: str | None) -> None:
+    def do_post_block_processing_cleanup(self) -> None:
```

Comment:
I don't like this function; it does too many things :)

1. it verifies that we have keyword params after '*'; this should be done in `state_parameter()` (IIUC, we've got what we need to catch this there)
2. it removes trailing whitespace from the param docstrings; this should be done in `state_parameter_docstring()`
3. it reformats the docstring and assigns it to the function

IMO, only 3. should be part of the post-processing. Validation should be done by the state machine; that's one of the features of a well-behaving state machine :)

Reply:
You know where to file an issue about _these_ problems ;)

Reply:
See gh-107550 for 2. Reworking 1. turned out to only make the code worse :(